src/Traits/GenericActionsTrait.php

```diff
@@ -222,8 +223,13 @@ public function object(string $uname): Response|null
         }
 
         $paths = $this->Publication->getViablePaths($entity->id);
+
         if (!empty($paths)) {
             return $this->redirect(['action' => 'fallback', $paths[0]['path']] + $params);
+        } else {
+            if ($entity->type === 'folders') {
+                throw new RecordNotFoundException();
+            }
```

Comment:
```suggestion
        } elseif ($entity->type === 'folders') {
            throw new RecordNotFoundException();
```

Reply:
👍  although it's a working progress so I kept the if alone because probably I have to add/modify stuff ;)